src/main/scala/com/mongodb/spark/api/java/sql/fields/MinKey.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.spark.api.java.sql.fields;
+
+/**
+ * A Java bean representing the Bson MinKey type
+ *
+ * @since 1.0
+ */
+public final class MinKey {
+    private int minKey;
+
+    /**
+     * Construct a new instance
+     */
+    public MinKey() {
+        this(1);
+    }
+
+    /**
+     * Construct a new instance
+     *
+     * @param minKey data representing the minKey
+     */
+    public MinKey(final int minKey) {
```

Comment:
As the struct has is `{ minKey: 1}` the field has to be called "minKey".  It also has to have the potential for a value - although it is ignored and only `BsonMinKey` is stored.